src/_shared/js/images.js

```diff
@@ -1,33 +1,34 @@
-export function addSourceset(responseJson) {
+import { write } from './dom';
 
-  let srcsetFragment = document.createDocumentFragment();
+export function addSourceset(responseJson) {
 
-  responseJson.forEach(source => {
+  return responseJson.map(source => {
 
     let highDef = document.createElement('source');
     highDef.media = `(min-width: ${source.minWidth}px) and
     (-webkit-min-device-pixel-ratio: 1.25),
     (min-width: ${source.minWidth}px) and (min-resolution: 120dpi)`;
     highDef.sizes = source.sizes;
-    highDef.srcset = `${window.location.protocol}${source.highDefSrcset}`;
+    highDef.srcset = `${source.hidpiSrcset}`;
 
     let lowDef = document.createElement('source');
     lowDef.media = `(min-width: ${source.minWidth}px)`;
     lowDef.sizes = source.sizes;
-    lowDef.srcset = `${window.location.protocol}${source.lowDefSrcset}`;
+    lowDef.srcset = `${source.lodpiSrcset}`;
 
-    srcsetFragment.appendChild(highDef);
-    srcsetFragment.appendChild(lowDef);
-  });
-    return srcsetFragment;
-}
+    return [highDef, lowDef];
+   });
+   }
 
 export function insertBetweenComments(sources) {
 
 	let pictures = Array.from(document.getElementsByTagName('picture'));
 
   return write(() => {
-		pictures.forEach((picture, index) => picture.insertBefore(sources[index], picture.firstChild));
+		pictures.forEach((picture, index) => {
+      picture.insertBefore(sources[index][0], picture.firstChild);
+      picture.insertBefore(sources[index][1], picture.firstChild);
+    });
```

Comment:
I don't think this works... There's only one `picture` tag with six sources, not three `picture` tags with two sources each. As a result this only inserts the first two `source` elements into the page and skips out the last four.